lightning/src/ln/channel.rs

```diff
@@ -8032,6 +8138,22 @@ impl<'a, 'b, 'c, ES: Deref, SP: Deref> ReadableArgs<(&'a ES, &'b SP, u32, &'c Ch
 			if iter.next().is_some() { return Err(DecodeError::InvalidValue) }
 		}
 
+		if let Some(malformed_htlcs) = malformed_htlcs {
+			for (htlc_id, failure_code, sha256_of_onion) in malformed_htlcs {
+				let dummy_htlc_idx = holding_cell_htlc_updates.iter().position(|htlc| {
+					if let HTLCUpdateAwaitingACK::FailHTLC { htlc_id: dummy_htlc_id, err_packet } = htlc {
+						let matches = htlc_id == *dummy_htlc_id;
+						if matches { debug_assert!(err_packet.data.is_empty()) }
+						matches
+					} else { false }
+				}).ok_or(DecodeError::InvalidValue)?;
+				let malformed_htlc = HTLCUpdateAwaitingACK::FailMalformedHTLC {
+					htlc_id, failure_code, sha256_of_onion
+				};
+				let _ = core::mem::replace(&mut holding_cell_htlc_updates[dummy_htlc_idx], malformed_htlc);
```

Comment:
I'd drop the `dummy_` prefix throughout and simply rename `htlc_id` to `malformed_htlc_id`. It's only a dummy htlc for ones that match.